diagnostics/include/performance_handler.h

```diff
@@ -146,6 +146,25 @@ class TimeRecorder {
   Recorder& recorder_;
 };
 
+// This is a wrapper around an unordered_map that records the durations of asynchronous actions.
+// It's useful when the timing being recorded can't tracked in a single scope, and there are
+// multiple outstanding requests that need timing, such as consensus slots.
+//
+// This type is not thread safe and always uses the non-atomic version of record.
```

Comment:
Why not make it thread-safe? In a single-threaded context, it does not have any overhead, but in other cases, like pre-processing, it will be used from multiple contexts.

Reply:
Good point. I will make it configurable via template parameter in a manner similar to the recorder.

Reply:
Fixed.